### PR TITLE
user updates with stripeCustomerObjId

### PR DESCRIPTION
**Pull request by daisyzarzo**
relates to https://github.com/bespoken/stripe-api/issues/3

### CODE REVIEW

File: lib/routes/payment.ts [hunk 1]
```diff
@@ -1,21 +1,139 @@
...
-    if (stripeErr) {
-        res.status(500).send({ error: stripeErr });
-    } else {
-        res.status(200).send({ success: stripeRes });
-    }
-};
-
-export const paymentApi = (app) => {
+export const paymentApi = async (app) => {
     app.get("/api", (req, res) => {
```
Comment (jperata) [inline lib/routes/payment.ts]: we can rename this to ping to use it for pingdom to ensure the server is up

File: lib/routes/payment.ts [hunk 2]
```diff
@@ -1,21 +1,139 @@
...
+            planIdToSubscribe = getPlanId(req.body.planToSubscribe);
+        } else {
+            Logger.error("There is no plan to subscribe like : " + req.body.planToSubscribe + " just Pro and Standard");
+            Returns.NotOkay(res, 500, "There is no plan to subscribe like : " + req.body.planToSubscribe).send();
+            return res;
+        }
+
+        if (req.body.userId && req.body.email && req.body.token &&
+            (!req.body.updateSubscribedPlan)) {
+            try {
```
Comment (jperata) [inline lib/routes/payment.ts]: you can extract all this block inside the if case into it's own createCustomer function, that way you can test that functionality

File: lib/routes/payment.ts [hunk 3]
```diff
@@ -1,21 +1,139 @@
...
+                // Create a Customer:
+                const customer = await Bluebird.resolve(
```
Comment (jperata) [inline lib/routes/payment.ts]: either stripe.customers.create is synchronous in which case this await Bluebird is not needed, or it is a promise in which case you can await the promise directly without Bluebird

File: lib/routes/payment.ts [hunk 4]
```diff
@@ -1,21 +1,139 @@
...
+
+            }
+
+        } else if (req.body.updateSubscribedPlan && req.body.stripeCustomerObjId
+            && req.body.stripeSubscribedPlanId) {
+
+            try {
+                const subscriptionItems: [any] = (await Bluebird.resolve(
+                    stripe.subscriptionItems.list({ subscription: req.body.stripeSubscribedPlanId }))).data;
+                await Bluebird.resolve(
```
Comment (jperata) [inline lib/routes/payment.ts]: same as the other bluebird comment

File: lib/routes/payment.ts [hunk 5]
```diff
@@ -1,21 +1,139 @@
...
+                    stripe.subscriptionItems.update(subscriptionItems[0].id,
+                        { plan: planIdToSubscribe }));
+
+                await Bluebird.resolve(registerStripeSubscribeToFireBase(req.body.stripeCustomerObjId,
```
Comment (jperata) [inline lib/routes/payment.ts]: same as the other bluebird comment

File: lib/routes/payment.ts [hunk 6]
```diff
@@ -1,21 +1,139 @@
...
+
+                Returns.NotOkay(res).send();
+                return res;
+
+            }
+
+        } else if (req.body.updateSubscribedPlan && req.body.stripeCustomerObjId
+            && req.body.stripeSubscribedPlanId) {
+
+            try {
```
Comment (jperata) [inline lib/routes/payment.ts]: you can extract all the stuff here to a updateCustomerSubscription function , that way you can test it

File: lib/routes/payment.ts [hunk 7]
```diff
@@ -1,21 +1,139 @@
...
+            Returns.NotOkay(res).send();
+            return res;
+        }
     });
 
     return app;
 };
+
+const subscribeUser = (stripeCustomerObjId: string, planId: string) => {
+    return Bluebird.resolve(stripe.subscriptions.create({
```
Comment (jperata) [inline lib/routes/payment.ts]: same as the other bluebird comments

File: lib/routes/payment.ts (hunk 3 above)
Comment (daisyzarzo) [reply lib/routes/payment.ts]: the bluebird let me handle the error messages  at the end  in the catch block

**Pull request merged**